middle_end/flambda2/types/env/cached_level.ml

```diff
@@ -147,15 +147,24 @@ let merge t1 t2 =
   in
   { names_to_types; aliases; symbol_projections }
 
-let remove_unused_closure_vars { names_to_types; aliases; symbol_projections }
-    ~used_closure_vars =
+let remove_unused_closure_vars_and_shortcut_aliases
+    { names_to_types; aliases; symbol_projections } ~used_closure_vars =
+  let canonicalise simple =
+    Simple.pattern_match simple
+      ~const:(fun _ -> simple)
+      ~name:(fun name ~coercion ->
+        Simple.apply_coercion_exn
+          (Aliases.get_canonical_ignoring_name_mode aliases name)
```

Comment:
Is it really okay to ignore name modes here?

Reply:
Yes. At this point all variables are going to have name mode `In_types`, so there won't be any cases were the canonical has mode `In_types` but there exists an alias at mode `Normal`.

Reply:
That makes sense, though it complicates my other suggestion; maybe it should be called `canonicalise_ignoring_name_modes`.

Reply:
(Also, it's worth explaining in a comment that all names have in-types mode, I think.)